From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nicolas Iooss <nicolas.iooss_linux@m4x.org>
Date: Sat, 18 Nov 2017 19:11:56 +0100
Subject: [PATCH] {BUG in 4.15-rc1} drivers/gpu/drm/i915/intel_pm: initialize
 trans_min

clang reports:

    drivers/gpu/drm/i915/intel_pm.c:4654:6: error: variable 'trans_min'
    is used uninitialized whenever 'if' condition is false
    [-Werror,-Wsometimes-uninitialized]
            if (INTEL_GEN(dev_priv) >= 10)
                ^~~~~~~~~~~~~~~~~~~~~~~~~
    drivers/gpu/drm/i915/i915_drv.h:2936:29: note: expanded from macro
    'INTEL_GEN'
    #define INTEL_GEN(dev_priv)     ((dev_priv)->info.gen)
                                    ^
    drivers/gpu/drm/i915/intel_pm.c:4657:19: note: uninitialized use
    occurs here
            trans_offset_b = trans_min + trans_amount;
                             ^~~~~~~~~
    drivers/gpu/drm/i915/intel_pm.c:4654:2: note: remove the 'if' if its
    condition is always true
            if (INTEL_GEN(dev_priv) >= 10)
            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    drivers/gpu/drm/i915/intel_pm.c:4639:20: note: initialize the
    variable 'trans_min' to silence this warning
            uint16_t trans_min, trans_y_tile_min;
                              ^
                               = 0
---
 drivers/gpu/drm/i915/intel_pm.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/i915/intel_pm.c b/drivers/gpu/drm/i915/intel_pm.c
index 3fe358db1276..17a17ebebe90 100644
--- a/drivers/gpu/drm/i915/intel_pm.c
+++ b/drivers/gpu/drm/i915/intel_pm.c
@@ -4898,7 +4898,7 @@ static void skl_compute_transition_wm(struct intel_crtc_state *cstate,
 {
 	struct drm_device *dev = cstate->base.crtc->dev;
 	const struct drm_i915_private *dev_priv = to_i915(dev);
-	uint16_t trans_min, trans_y_tile_min;
+	uint16_t trans_min = 0, trans_y_tile_min;
 	const uint16_t trans_amount = 10; /* This is configurable amount */
 	uint16_t trans_offset_b, res_blocks;
 
-- 
